File: libs/libmpsse/src/examples/bitbang.c
#include "mpsse_config.h"
#include <stdio.h>
#include <stdlib.h>
#include <mpsse.h>

int main(void)
{
	struct mpsse_context *io = NULL;
	int i = 0, retval = EXIT_FAILURE;

	io = MPSSE(BITBANG, 0, 0);

	if(io && io->open)
	{
		for(i=0; i<10; i++)
		{
			PinHigh(io, 0);
			printf("Pin 0 is: %d\n", PinState(io, 0, -1));
			sleep(1);
			
			PinLow(io, 0);
			printf("Pin 0 is: %d\n", PinState(io, 0, -1));
			sleep(1);
		}

		retval = EXIT_SUCCESS;
	}
	else
	{
		printf("Failed to open MPSSE: %s\n", ErrorString(io));
	}

	Close(io);

	return retval;
}

File: libs/libmpsse/src/examples/bitbang.py
#!/usr/bin/env python2

from mpsse import *
from time import sleep

# Open mpsse in bitbang mode
io = MPSSE(BITBANG);

# Set pin 0 high/low and read back pin 0's state 10 times
for i in range(0, 10):

    io.PinHigh(0)
    print "Pin 0 is:", io.PinState(0)
    sleep(1)

    io.PinLow(0)
    print "Pin 0 is:", io.PinState(0)
    sleep(1)

io.Close()
